rust/storage/src/hummock/mod.rs

```diff
@@ -228,63 +257,53 @@ impl StateStore for HummockStorage {
                 self.stats.get_shared_buffer_hit_counts.inc();
                 return Ok(v.into_put_value().map(StorageValue::from));
             }
+            let internal_key = key_with_epoch(key.to_vec(), epoch);
 
             let mut table_counts = 0;
             for level in &version.levels() {
+                let mut tables = self
+                    .local_version_manager
+                    .pick_few_tables(&level.table_ids)
+                    .await?;
                 match level.level_type() {
                     LevelType::Overlapping => {
-                        let tables = bloom_filter_sstables(
-                            self.local_version_manager
-                                .pick_few_tables(&level.table_ids)
-                                .await?,
-                            key,
-                            self.stats.clone(),
-                        )?;
-                        table_counts += tables.len();
-                        table_iters.extend(tables.into_iter().map(|table| {
-                            Box::new(SSTableIterator::new(table, self.sstable_store.clone()))
-                                as BoxedHummockIterator
-                        }))
+                        tables.reverse();
+                        for table in tables {
+                            table_counts += 1;
+                            if let Some(v) = self.get_from_table(table, &internal_key, key).await? {
+                                return Ok(Some(StorageValue::from(v)));
+                            }
+                        }
```

Comment:
Even now files in L0 are not guaranteed to be in epoch order, because new version generated by a compaction would have sorted L0 according to key_range of file. @soundOfDestiny 
Though it doesn't affect the correctness of the code here.

Reply:
> Even now files in L0 are not guaranteed to be in epoch order, because new version generated by a compaction would have sorted L0 according to key_range of files. @soundOfDestiny
> Though it doesn't affect the correctness of the code here.

yes

Reply:
fixed in https://github.com/singularity-data/risingwave/pull/2062